cmd/kubeadm/app/cmd/phases/controlplane.go

```diff
@@ -18,185 +18,155 @@ package phases
 
 	schedulerControlplaneLongDesc = fmt.Sprintf(normalizer.LongDesc(`
 		Generates the static Pod manifest file for the scheduler and saves it into %s file.
-		`+cmdutil.AlphaDisclaimer), kubeadmconstants.GetStaticPodFilepath(kubeadmconstants.KubeScheduler, kubeadmconstants.GetStaticPodDirectory()))
+		`), kubeadmconstants.GetStaticPodFilepath(kubeadmconstants.KubeScheduler, kubeadmconstants.GetStaticPodDirectory()))
 )
 
-// NewCmdControlplane returns main command for Controlplane phase
-func NewCmdControlplane() *cobra.Command {
-	cmd := &cobra.Command{
-		Use:   "controlplane",
+type controlPlaneData interface {
+	Cfg() *kubeadmapi.InitConfiguration
+	ManifestDir() string
+}
+
+// NewControlPlanePhase creates a kubeadm workflow phase that implements bootstrapping the control plane.
+func NewControlPlanePhase(cfg *kubeadmapi.InitConfiguration) workflow.Phase {
+	phase := workflow.Phase{
+		Name:  "control-plane",
 		Short: "Generates all static Pod manifest files necessary to establish the control plane",
 		Long:  cmdutil.MacroCommandLongDescription,
+		Run:   runControlPlane,
 	}
-
 	manifestPath := kubeadmconstants.GetStaticPodDirectory()
-	cmd.AddCommand(getControlPlaneSubCommands(manifestPath, "")...)
-	return cmd
+	subPhases := getControlPlaneSubPhases(cfg, manifestPath)
+	for _, p := range subPhases {
+		phase.AppendPhase(*p)
+	}
+	return phase
 }
 
-// getControlPlaneSubCommands returns sub commands for Controlplane phase
-func getControlPlaneSubCommands(outDir, defaultKubernetesVersion string) []*cobra.Command {
-
-	cfg := &kubeadmapiv1beta1.InitConfiguration{}
-
-	// This is used for unit testing only...
-	// If we wouldn't set this to something, the code would dynamically look up the version from the internet
-	// By setting this explicitly for tests workarounds that
-	if defaultKubernetesVersion != "" {
-		cfg.KubernetesVersion = defaultKubernetesVersion
-	}
+// runControlPlane executes the control plane phase
+func runControlPlane(c workflow.RunData) error {
+	glog.V(1).Infof("[control-plane] bootstraping the control plane")
+	glog.V(1).Infof("[control-plane] creating static pod manifests")
+	return nil
+}
 
-	// Default values for the cobra help text
-	kubeadmscheme.Scheme.Default(cfg)
+// getControlPlaneSubCommands returns sub commands for Controlplane phase
+func getControlPlaneSubPhases(cfg *kubeadm.InitConfiguration, outDir string) []*workflow.Phase {
 
-	var cfgPath, featureGatesString string
-	var subCmds []*cobra.Command
+	const (
+		phaseAPIServer         = "apiserver"
+		phaseControllerManager = "controller-manager"
+		phaseScheduler         = "scheduler"
+	)
+	var subPhases []*workflow.Phase
 
 	subCmdProperties := []struct {
-		use      string
+		name     string
 		short    string
 		long     string
 		examples string
-		cmdFunc  func(outDir string, cfg *kubeadmapi.InitConfiguration) error
+		cmdFunc  func(outDir string, cfg *kubeadm.InitConfiguration) error
```

Comment:
If you check to PR I linked, I'm proposing an approach that will allow you to get rid of all the Create* wrappers in favor of createStaticPodFiles